### PR TITLE
Select radius interface

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Define for an interface if you want to have radius available

# Impacts
Radius Workflow

# Delete branch after merge
YES

# NEWS file entries
## New Features
* Define on interface if radius must listen on.

### CODE REVIEW

File: lib/pf/Switch.pm [hunk 1]
```diff
@@ -3112,12 +3112,22 @@ Takes into account the active/active clustering and centralized deauth
 =cut
 
 sub deauth_source_ip {
-    my ($self) = @_;
+    my ($self,$dst_ip) = @_;
+    my $chi = pf::CHI->new(namespace => 'route_int');
+    my $int = $chi->compute($dst_ip, sub {
```
Comment (julsemaan) [inline lib/pf/Switch.pm]: We need to validate that the result of this is a valid interface configured in PacketFence

File: lib/pf/services/manager/keepalived.pm [hunk 2]
```diff
@@ -59,7 +60,7 @@ sub generateConfig {
 
     $tags{'vrrp'} = '';
     $tags{'mysql_backend'} = '';
-    my @ints = uniq(@listen_ints,@dhcplistener_ints, map { $_->{'Tint'} } @portal_ints);
+    my @ints = uniq(@listen_ints,@dhcplistener_ints, map { $_->{'Tint'} } @portal_ints, @radius_ints);
```
Comment (julsemaan) [inline lib/pf/services/manager/keepalived.pm]: wrap the map in parentheses so its clearer what you are mapping and to make sure it only applies to what you want:
`uniq(@listen_ints,@dhcplistener_ints, (map { $_->{'Tint'} } @portal_ints, @radius_ints));`

**Comment by julsemaan** [comment]
I'll find some time to test it a bit

File: lib/pf/Switch.pm [hunk 3]
```diff
@@ -3112,12 +3112,32 @@ Takes into account the active/active clustering and centralized deauth
...
+                           );
+    if (defined($Config{ 'interface ' . $int })) {
+        if($cluster_enabled){
+            return isenabled($Config{active_active}{centralized_deauth}) ? pf::cluster::cluster_ip($int) : pf::cluster::current_server->{"interface $int"}->{ip};
+        }
+        else {
+            return $Config{ 'interface ' . $int }{'vip'} || $Config{ 'interface ' . $int }{'ip'}
+        }
+    } else {
+        $logger->warn("Interface $int has not been found in the configuration, using the management interface");
```
Comment (julsemaan) [inline lib/pf/Switch.pm:3136]: that logger is not defined

**Comment by julsemaan** [comment]
If I add radius as an additional listening daemon on a registration network, it doesn't add it in pf.conf and makes the interface become of the 'Other' type in the admin

**Comment by julsemaan** [comment]
Also, there is no way to put an interface in type 'radius' from the admin, its only possible to put it as an additionnal listening daemon

**Comment by julsemaan** [comment]
Other than my 2 comments above it seems to do what it says

File: lib/pf/ConfigStore/Network.pm [hunk 4]
```diff
@@ -142,10 +142,11 @@ sub cleanupBeforeCommit {
     my $config = $self->cachedConfig;
     unless ( $config->SectionExists($id) ) {
         # Set default values when creating a new network
+        my @types = split(',',$network->{type});
```
Comment (julsemaan) [inline lib/pf/ConfigStore/Network.pm]: split while cleaning up spaces

File: lib/pf/Switch.pm [hunk 5]
```diff
@@ -3113,12 +3113,33 @@ Takes into account the active/active clustering and centralized deauth
...
-    if($cluster_enabled){
-        return isenabled($Config{active_active}{centralized_deauth}) ? pf::cluster::management_cluster_ip() : pf::cluster::current_server->{management_ip};
-    }
-    else {
-        return $management_network->tag('vip') || $management_network->tag('ip');
+    my ($self,$dst_ip) = @_;
+    my $logger = $self->logger();
+    my $chi = pf::CHI->new(namespace => 'route_int');
+    my $int = $chi->compute($dst_ip, sub {
+                                         my @interface_src = split(" ", `LANG=C sudo ip route get $dst_ip`);
```
Comment (julsemaan) [inline lib/pf/Switch.pm]: Use pf_run ?

File: lib/pf/services/manager/radsniff.pm [hunk 6]
```diff
@@ -40,4 +44,18 @@ has '+launcher' => (
 
 has startDependsOnServices => ( is => 'ro', default => sub { [qw(collectd)] } );
 
+sub make_filter {
```
Comment (julsemaan) [inline lib/pf/services/manager/radsniff.pm]: Missing perldoc

File: lib/pfconfig/namespaces/interfaces/radius_ints.pm [hunk 7]
```diff
@@ -0,0 +1,64 @@
+package pfconfig::namespaces::interfaces::radius_ints;
```
Comment (julsemaan) [inline lib/pfconfig/namespaces/interfaces/radius_ints.pm:1]: perldoc for the two subs in this module

**Review by julsemaan — CHANGES_REQUESTED** [review]
A few changes here and there and we're good to merge

**Comment by extrafu** [comment]
Bump

**Pull request merged**